### PR TITLE
Add Localstack container support

### LINKED ISSUE #33: Feature Request:  Localstack container
Localstack is a mock of AWS services.  There is a Localstack available in the testcontainers-java library: https://github.com/testcontainers/testcontainers-java/blob/master/modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java

It would be very useful if this were directly available in the python module.  My teams current workaround is to use the DockerCompose object with this docker-compose.yaml file:

```
version: '2.1'

services:
  localstack:
    image: localstack/localstack
    ports:
      - "4567-4597:4567-4597"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES="kinesis,lambda"
      - DEBUG=true
      - USE_SSL=true
      - DOCKER_HOST=unix:///var/run/docker.sock
```

### PR DESCRIPTION
**Pull request by ImFlog**
This PR adds an equivalent to the testcontainers-java Localstack container.
This is simpler than the Java version because a simple edge port was introduced in version 0.11.0. 
This should fix #33.

### CODE REVIEW

File: testcontainers/localstack.py
Comment (hermanschaaf) [inline testcontainers/localstack.py:27]:
```suggestion
        dynamo_endpoint = localstack.get_endpoint_override()
```

When running this locally, I found `get_service_endpoint` to not be defined. Should this be `get_endpoint_override`?

File: testcontainers/localstack.py
Comment (hermanschaaf) [inline testcontainers/localstack.py:57]:
```suggestion
    def _get_ready_log(self, timeout=60):
```

Using this with only the S3 service often takes longer than 30 seconds, at least on my machine. 60 seconds seems to be enough.

It might also be nice to be able to override this timeout as a parameter to the `start` method

File: testcontainers/localstack.py
Comment (hermanschaaf) [inline testcontainers/localstack.py:38]: Is this line necessary? Does localstack spin up new containers in some circumstances?

Comment (ImFlog) [reply testcontainers/localstack.py:38]: For lambdas it needs to spin up containers yes. Most of the time I use this so I added it.
I can remove it or create a function doing this, WDYT ?

Comment (hermanschaaf) [reply testcontainers/localstack.py:38]: Ah, interesting. Would it be an idea to only set the volume mapping if the `lambda` service is requested? In some CI environments (such as ours) it would not be desirable for additional containers to be created, as they might not get cleaned up afterwards

Comment (ImFlog) [reply testcontainers/localstack.py:38]: I am not sure that only lambda needs this to be honest. Also some other ENV vars can change this behavior and make the lambda run locally instead of new containers.
So for now I think it's better if we let the users chose when to put this volume mapping according to what's span.
I will remove it.

File: testcontainers/localstack.py
Comment (hermanschaaf) [inline testcontainers/localstack.py:57]: Out of curiosity: why is there a slash here? (As if the string should be `Ready\.` ?)

Comment (ImFlog) [reply testcontainers/localstack.py]: Without it PEP8 was yelling so I did this.
Your comment made me realize that indeed this was strange and now It's clearer :)

**Comment by ImFlog** [comment]
The comment were super useful, thanks @hermanschaaf !
Now let's wait for a maintainer review (I think it's @tillahoffmann).

**Comment by ImFlog** [comment]
Anyone having the right on the repository willing to review this ?
AFAIK there is at least 1 company (mine) and @hermanschaaf that are interested in this :)

**Comment by Iprogrammerr** [comment]
My company is interested also. We have a lot of small python apps that interact with aws environment. We are already using testcontainers for our db tests and it's great :)

File: testcontainers/localstack.py
Comment (tillahoffmann) [inline testcontainers/localstack.py:44]:
```suggestion
        return self.with_env('SERVICES', ','.join(services))
```

Small tweak to allow chaining of `with_*` calls.

File: testcontainers/localstack.py
Comment (tillahoffmann) [inline testcontainers/localstack.py:56]: Is this function being used elsewhere? If not, let's inline it within `start`.

File: testcontainers/localstack.py
Comment (tillahoffmann) [inline testcontainers/localstack.py:46]: Can we add a docstring here? Do you think it makes sense to use the same naming (`get_url`) as in the `ElasticSearchContainer`?

https://github.com/testcontainers/testcontainers-python/blob/56e6b88d151509c83487f593efa8b12654a47673/testcontainers/elasticsearch.py#L43-L46

Comment (ImFlog) [reply testcontainers/localstack.py:46]: In the boto3 library, when overriding the url of a client a kwarg `endpoint_url=XXX` is used.
Maybe I can rename to that so it easier to understand when using the testcontainer ?
I can also change to get_url if you want.

Comment (tillahoffmann) [reply testcontainers/localstack.py]: Yes, let's go with `get_url` for consistency.

**Comment by shashanksingh** [comment]
Heya, my org would also love to use it, we are waiting for this merge to write e2e tests. Please do tell me if I can help anyway.

**Review by tillahoffmann — APPROVED** [review]

**Pull request merged**
